### PR TITLE
Scroll dev 1115

### PR DESCRIPTION
**Pull request by lispc**
TODO:
  
1. multi block for public_input circuit / tx circuit. 
2. comment tx->block_num lookup in End(Inner)Blcok

### CODE REVIEW

File: bus-mapping/src/circuit_input_builder.rs [hunk 1]
```diff
@@ -240,17 +309,69 @@ impl<'a> CircuitInputBuilder {
...
+                geth_step.pc.0,
+                geth_step.gas.0,
+                state_ref.block_ctx.rwc.0,
+                state_ref.call().map(|c| c.call_id).unwrap_or(0),
+                if geth_step.op.is_push() {
+                    match geth_step.stack.last() {
+                        Ok(w) => format!("{:?}", w),
+                        Err(_) => "".to_string(),
+                    }
+                } else if geth_step.op.is_call6() {
```
Comment (DreamWuGit) [inline bus-mapping/src/circuit_input_builder.rs:350]: oh, `is_call6` and `is_call7` means pop 6 or 7 item from stack after think a while  ,  rename to `is_call_callcode` `is_static_delegate_call` seems  better reading

File: bus-mapping/src/rpc.rs [hunk 2]
```diff
@@ -100,6 +100,17 @@ impl<P: JsonRpcClient> GethClient<P> {
             .await
             .map_err(|e| Error::JSONRpcError(e.into()))
     }
+    /// ..
+    pub async fn get_tx_by_hash(&self, hash: H256) -> Result<Transaction, Error> {
```
Comment (DreamWuGit) [inline bus-mapping/src/rpc.rs:104]: this rpc call targets on L2geth node right ?

Comment (lispc) [reply bus-mapping/src/rpc.rs:104]: both l1 and l2 work. This rpc is intended to be used to mock prove single tx.

**Comment by lispc** [comment]
close in favor of https://github.com/scroll-tech/zkevm-circuits/pull/229

**Pull request closed without merging**